compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -112,6 +112,10 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
           if (tp.isTerm)
             ctx.error(i"splice outside quotes", pos)
           tp
+        case tp: TypeRef if tp.prefix.derivesFrom(defn.QuotedTypeClass) && tp.name.toString == "T" =>
```

Comment:
It's fragile and un-idiomatic to compare a name. If we have to do this, the name needs to be forge-proof (i.e. use `$` characters in it, and there should be a fat comment in QuotedTypeClass that the type parameter name cannot be changed. 

But I hope we do not need to do that, and that this works instead:
```scala
case tp: TypeRef if tp.symbol == defn.QuotedTypeClass.typeParams.head
```

Reply:
It worked with that change